ddtrace/contrib/anthropic/_streaming.py

```diff
@@ -0,0 +1,322 @@
+    chunk_type = getattr(chunk, "type", "")
+    transformation = TRANSFORMATIONS_BY_BLOCK_TYPE.get(chunk_type)
+    if transformation is not None:
+        message = transformation(chunk, message)
+
+    return message
+
+
+def _on_message_start_chunk(chunk, message):
+    # this is the starting chunk of the message
+    if getattr(chunk, "type", "") != "message_start":
+        return message
+
+    chunk_message = getattr(chunk, "message", "")
+    if chunk_message:
+        content_text = ""
+        contents = getattr(chunk.message, "content", [])
+        for content in contents:
+            if content.type == "text":
+                content_text += content.text
+                content_type = "text"
+            elif content.type == "image":
+                content_text = "([IMAGE DETECTED])"
+                content_type = "image"
+            message["content"].append({"text": content_text, "type": content_type})
+
+        chunk_role = getattr(chunk_message, "role", "")
+        chunk_usage = getattr(chunk_message, "usage", "")
+        chunk_finish_reason = getattr(chunk_message, "stop_reason", "")
+        if chunk_role:
+            message["role"] = chunk_role
+        if chunk_usage:
+            message["usage"] = {}
+            message["usage"]["input_tokens"] = getattr(chunk_usage, "input_tokens", 0)
+            message["usage"]["output_tokens"] = getattr(chunk_usage, "output_tokens", 0)
+        if chunk_finish_reason:
+            message["finish_reason"] = chunk_finish_reason
+    return message
+
+
+def _on_content_block_start_chunk(chunk, message):
+    # this is the start to a message.content block (possibly 1 of several content blocks)
+    if getattr(chunk, "type", "") != "content_block_start":
+        return message
+
+    message["content"].append({"type": "text", "text": ""})
+    return message
+
+
+def _on_content_block_delta_chunk(chunk, message):
+    # delta events contain new content for the current message.content block
+    if getattr(chunk, "type", "") != "content_block_delta":
+        return message
+
+    delta_block = getattr(chunk, "delta", "")
+    chunk_content = getattr(delta_block, "text", "")
+    if chunk_content:
+        message["content"][-1]["text"] += chunk_content
+    return message
+
+
+def _on_message_delta_chunk(chunk, message):
+    # message delta events signal the end of the message
+    if getattr(chunk, "type", "") != "message_delta":
+        return message
+
+    delta_block = getattr(chunk, "delta", "")
+    chunk_finish_reason = getattr(delta_block, "stop_reason", "")
+    if chunk_finish_reason:
+        message["finish_reason"] = chunk_finish_reason
+        message["content"][-1]["text"] = message["content"][-1]["text"].strip()
+
+    chunk_usage = getattr(chunk, "usage", {})
+    if chunk_usage:
+        message_usage = message.get("usage", {"output_tokens": 0, "input_tokens": 0})
+        message_usage["output_tokens"] += getattr(chunk_usage, "output_tokens", 0)
```

Comment:
I see from the streaming API [docs](https://docs.anthropic.com/en/api/messages-streaming#basic-streaming-request) that the message_start content block will include `output_tokens = 1`, but also the message_delta chunk will include the correct output token count. It looks like we are adding that 1 (from message_start) to this which may be correct. Can we just rely on the output token information from message_delta and input_token information from message_start?

Reply:
I'm not sure if the output + 1 is correct or just output.

Reply:
okay i've updated as requested